components/post_view/post_add_bot_teams_channels/post_add_bot_teams_channels.jsx

```diff
@@ -0,0 +1,207 @@
+
+import FormattedMarkdownMessage from 'components/formatted_markdown_message';
+import GenericTeamChannelProvider from 'components/suggestion/generic_team_channel_provider.jsx';
+import MenuActionProvider from 'components/suggestion/menu_action_provider';
+import AutocompleteSelector from 'components/widgets/settings/autocomplete_selector';
+import PostEmoji from 'components/post_emoji';
+
+import {localizeMessage} from 'utils/utils.jsx';
+
+export default class PostAddBotTeamsChannels extends React.PureComponent {
+    static propTypes = {
+
+        /*
+        * Post containing the user id of the bot
+        */
+        post: PropTypes.object.isRequired,
+
+        /*
+        * List of teams available to the current user
+        */
+        teams: PropTypes.arrayOf(PropTypes.object).isRequired,
+
+        actions: PropTypes.shape({
+
+            /*
+            * Function to add members to channel
+            */
+            addChannelMember: PropTypes.func.isRequired,
+
+            /*
+            * Function to add user to team
+            */
+            addUserToTeam: PropTypes.func.isRequired,
+        }).isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {addMode: false};
+    }
+
+    handleAddButton = () => {
+        this.setState({addMode: true});
+    }
+
+    handleSelectedTeam = (selected) => {
+        this.setState({selectedTeamName: selected.text, team: selected});
+    }
+
+    handleSelectedChannel = (selected) => {
+        this.setState({selectedChannelName: selected.display_name, channel: selected});
+
+        return this.props.actions.addUserToTeam(this.state.team.value, this.props.post.user_id).then(
+            () => this.props.actions.addChannelMember(this.state.channel.id, this.props.post.user_id)).then(
+            () => {
+                // Check to see if we've already added the bot to the channel, it doesn't need to show up twice
+                var currentAddedChannels = this.props.post.props.addedChannels;
+                if (currentAddedChannels) {
+                    var channelAlreadyExists = currentAddedChannels.find((c) => c.id === this.state.channel.id);
+                    if (!channelAlreadyExists) {
+                        currentAddedChannels = currentAddedChannels.concat([{
+                            id: this.state.channel.id,
+                            name: this.state.selectedChannelName,
+                        }]);
+                    }
+                } else {
+                    currentAddedChannels = [{
+                        id: this.state.channel.id,
+                        name: this.state.selectedChannelName,
+                    }];
+                }
+
+                const post = {
+                    id: this.props.post.id,
+                    props: {
+                        addedChannels: currentAddedChannels,
+                    },
+                };
+
+                return Client4.updatePost(post);
```

Comment:
Can you pass in an action as a prop to do this instead of calling the web client directly? There's some extra bits that need to happen when using the web client that is taken care of by using the actions from mattermost-redux. I'd also suggest using patch vs update to make sure you're only changing the fields you want to

Reply:
Yeah I wasn't happy about this either, @hmhealey mentioned this to me as well. I'm using patchPost now.